File: Lab2/Lab23/app/src/main/java/com/example/lab23/MainActivity.java
package com.example.lab23;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
    TextView txtWelcome;

    Button btnLogout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        txtWelcome = findViewById(R.id.txtWelcome);
        btnLogout = findViewById(R.id.btnLogout);

        Intent intent = getIntent();

        Bundle bundle = intent.getBundleExtra("myPackage");
        if (bundle != null) {
            String username = bundle.getString("username");
            String password = bundle.getString("password");
            txtWelcome.setText("Welcome: " + username + "\n" + "Your password is: " + password);
        }



        btnLogout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SignInActivity.class);
                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
                startActivity(intent);
                finish(); // Close MainActivity
            }
        });

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: Lab45/app/src/main/java/com/example/lab45/MainActivity.kt
package com.example.lab45

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        
        val buttonLab4 = findViewById<Button>(R.id.buttonLab4)
        val buttonLab5 = findViewById<Button>(R.id.buttonLab5)
        
        buttonLab4.setOnClickListener {
            startActivity(Intent(this, Lab4MainActivity::class.java))
        }
        
        buttonLab5.setOnClickListener {
            startActivity(Intent(this, Lab5MainActivity::class.java))
        }
    }
}